### PR TITLE
Event wise rad max

### PR DESCRIPTION
**Pull request by maxnoe**
**Description**

    
Instead of evaluating rad_max at the bin center of the given counts
energy axis, we now use the individual event energy.

By making the WobbleRegionsFinder work properly with PointSkyRegion and
adding a check for overlapping regions in the rad_max case, we can now
perform the classical data reduction for point sky analyses.

The ReflectedRegionsFinder does not work and probably cannot be made to
work with PointSkyRegion, since it uses the extent of the previous
region to place the next. This is obviously not possible for
PointSkyRegion.

This simplified the code in `make_counts_rad_max` and `make_counts_off_rad_max` quite significantly, there is now only a loop over the off regions and not over the reco energy bins anymore.

### CODE REVIEW

**Comment by adonath** [comment]
One preliminary comment: I think it could make sense to move the overlapping method to `gammapy.utils.region`, this is where we keep functionality, that might end up in `regions` at some point. Ideally we'd have `region.overlaps(region_other, was)`, but this is not supported. So  my proposed signature would be something like `is_region_overlapping(region, region_other, wcs)` as helper method on `gammapy.utils.region`, I can easily imagine it will be helpful in other situations as well.

**Comment by maxnoe** [comment]
@adonath Are you talking about the more general check in the `WobbleRegionsFinder`? Because the new method here is rather specific to rad max.

File: gammapy/makers/utils.py [hunk 1]
```diff
@@ -472,26 +474,51 @@ def make_counts_rad_max(geom, rad_max, events):
...
+    if rad_max.data.shape == (1, 1):
+        rad_max_for_events = rad_max.quantity[0, 0]
+    else:
+        rad_max_for_events = rad_max.evaluate(energy=events.energy, offset=offset)
+
+    selected = separation <= rad_max_for_events
+    return events.select_row_subset(selected)
+
+
+def are_regions_overlapping_rad_max(regions, rad_max, offset):
```
Comment (adonath) [inline gammapy/makers/utils.py:500]: OK, this specific implementation might remain. The region based alternative would be to create a list of `CircleSkyRegions` and then use again `np.any([regions_overlap(reg_1, reg_2, wcs=wcs) for reg_1, reg_2 in combinations(regions, 2)])`

However `regions_overlap` could implement the short cut for `CircleSkyRegion`, based on the separation. But of course the purely region based method will be slower...not sure it matters...

Comment (adonath) [reply gammapy/makers/utils.py:500]: On the other hand: having some functionality to create a list of circular regions from a `RadMax2D` object might be useful e.g. for plotting as well..

**Review by adonath — APPROVED** [review]
Thanks @maxnoe, I think this PR is ready to merge. The `regions` related comments we can solve in follow up PRs if useful...

**Comment by adonath** [comment]
@cosimoNigro Can you maybe add your review here?

**Review by cosimoNigro — APPROVED** [review]

File: gammapy/makers/utils.py [hunk 2]
```diff
@@ -472,26 +474,51 @@ def make_counts_rad_max(geom, rad_max, events):
...
 
-    counts = RegionNDMap.from_geom(geom, data=np.asarray(counts_list))
 
-    return counts
+def apply_rad_max(events, rad_max, position):
+    '''Apply the RAD_MAX cut to the event list for given region'''
+    offset = position.separation(events.pointing_radec)
+    separation = position.separation(events.radec)
+
+    if rad_max.data.shape == (1, 1):
```
[the remark this answers is not in the document]
Comment (maxnoe) [reply gammapy/makers/utils.py:491]: https://github.com/gammapy/gammapy/issues/3835

**Comment by maxnoe** [comment]
I adapted the tests for the updated magic DL3 data. The RAD_MAX in the files is so, that only 1 off region is possible.

**Review by registerrier — APPROVED** [review]
Thanks @maxnoe . This looks good!

I have left few inline comments.

File: gammapy/makers/utils.py [hunk 3]
```diff
@@ -472,26 +474,53 @@ def make_counts_rad_max(geom, rad_max, events):
...
+    ])
+
+
+    # evaluate fails with a single bin somewhere trying to interpolate
+    if rad_max.data.shape == (1, 1):
+        rad_max_at_offset = rad_max.quantity[0, 0]
+    else:
+        rad_max_at_offset = rad_max.evaluate(offset=offset)
+
+    return np.any(separations[np.newaxis, :] < (2 * rad_max_at_offset))
```
[the remark this answers is not in the document]
Comment (maxnoe) [reply gammapy/makers/utils.py:519]: Good point.

File: gammapy/makers/utils.py [hunk 4]
```diff
@@ -519,44 +548,41 @@ def make_counts_off_rad_max(
...
+        center=events.pointing_radec,
+        region=on_geom.region,
+        exclusion_mask=exclusion_mask,
+    )
 
-    # we have to define an ON region and a region finder for each energy bin
-    for i, rad in enumerate(rad_max):
+    if len(off_regions) == 0:
+        # counts_off=None, acceptance_off=0
+        return None, RegionNDMap.from_geom(on_geom, data=0)
```
[the remark this answers is not in the document]
Comment (maxnoe) [reply gammapy/makers/utils.py:561]: Adding the test for this, it turned out that the exclusion mask check in the `WobbleRegionsFinder` did not work for `PointSkyRegion`. This is now fixed.

File: gammapy/makers/utils.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (maxnoe) [reply gammapy/makers/utils.py:529]: This is now implemented and tested

**Comment by maxnoe** [comment]
@registerrier Thanks, that revealed an important issue (exclusion mask not working for `PointSkyRegion`, which is now fixed and tested.

Can you review again?

**Review by registerrier — DISMISSED** [review]
Thanks @maxnoe ! 

Simply remove the meta definition from this PR and we merge.

**Comment by maxnoe** [comment]
Done, sorry again, those files somehow kept creeping in...

**Comment by registerrier** [comment]
Thanks @maxnoe . Merging now.

**Pull request merged**